src/pymodulon/plotting.py

```diff
@@ -1442,9 +1475,195 @@ def plot_explained_variance(ica_data, pc=True, ax=None):
+    scatter_kwargs=None,
+    ax_font_kwargs=None,
+    legend_kwargs=None,
+    savefig_kwargs=None,
+):
+    """
+    Compare the overlaps between iModulons and their linked regulons
+
+    Parameters
+    ----------
+    ica_data: ~pymodulon.core.IcaData
+        :class:`~pymodulon.core.IcaData` object
+    imodulons: list, optional
+        List of iModulons to plot
+    cat_column: str, optional
+        Column in the `imodulon_table` that stores the category of each iModulon
+    size_column: str, optional
+        Column in the `imodulon_table` that stores the size of each iModulon
+    scale: float, optional (default: 1)
+        Value used to scale the size of each point
+    reg_only: bool (default: True)
+        Only plot iModulons with an entry in the `regulator` column of the
+        `imodulon_table`
+    xlabel: str, optional
+        Custom x-axis label (default: "# shared genes/Regulon size")
+    ylabel: str, optional
+        Custom y-axis label (default: "# shared genes/iModulon size")
+    vline: float, optional (default: 0.6)
+        Draw a dashed vertical line
+    hline: float, optional (default: 0.6)
+        Draw a dashed horizontal line
+    ax: ~matplotlib.axes.Axes, optional
+        Axes object to plot on, otherwise use current Axes
+    savefig: str or bool, optional
+        Save figure to provided path
+    scatter_kwargs: dict, optional
+        Additional keyword arguments passed to :func:`matplotlib.pyplot.scatter`
+    ax_font_kwargs: dict, optional
+        Additional keyword arguments for axes labels
+    legend_kwargs: dict, optional
+        Additional keyword arguments passed to :func:`matplotlib.pyplot.legend`
+    savefig_kwargs: dict, optional
+        Additional keyword arguments passed to :func:`matplotlib.pyplot.savefig`
+
+    Returns
+    -------
+    ax: ~matplotlib.axes.Axes
+        :class:`~matplotlib.axes.Axes` containing the line plot
+    """
+
+    # Set up axis
+    if not ax:
+        fig, ax = plt.subplots(figsize=(5, 5))
+    else:
+        fig = ax.figure
+
+    # Handle kwargs
+    if scatter_kwargs is None:
+        scatter_kwargs = {}
+    if ax_font_kwargs is None:
+        ax_font_kwargs = {}
+    if legend_kwargs is None:
+        legend_kwargs = {}
+
+    # Set axes labels
+    if xlabel is None:
+        xlabel = "$\\frac{\\mathrm{\\#\\;shared\\;genes}}{\\mathrm{Regulon\\;size}}$"
+        xlabelscale = 2
+    else:
+        xlabelscale = 1
+
+    if ylabel is None:
+        ylabel = "$\\frac{\\mathrm{\\#\\;shared\\;genes}}{\\mathrm{iModulon\\;size}}$"
+        ylabelscale = 2
+    else:
+        ylabelscale = 1
+
+    # Select iModulons
+    if imodulons is None:
+        imodulon_table = ica_data.imodulon_table
+        if reg_only:
+            imodulon_table = imodulon_table[imodulon_table.regulator.notnull()]
+    else:
+        imodulon_table = ica_data.imodulon_table.loc[imodulons]
+
+    # Select category column
+    reg_table = imodulon_table.copy()
```

Comment:
You should be able to get rid of this extra copy  in line 1584 by changing line 1577 to
` reg_table = ica_data.imodulon_table.copy()` 
and change rename `imodulon_table` in lines 1579 and 1581 to `reg_table` since both those functions return a copy of the imodulon_table anyways.